lightning-background-processor/src/lib.rs

```diff
@@ -886,7 +886,16 @@ mod tests {
 		fn disconnect_socket(&mut self) {}
 	}
 
-	type ChannelManager = channelmanager::ChannelManager<Arc<ChainMonitor>, Arc<test_utils::TestBroadcaster>, Arc<KeysManager>, Arc<KeysManager>, Arc<KeysManager>, Arc<test_utils::TestFeeEstimator>, Arc<DefaultRouter<Arc<NetworkGraph<Arc<test_utils::TestLogger>>>, Arc<test_utils::TestLogger>, Arc<Mutex<TestScorer>>, (), TestScorer>>, Arc<test_utils::TestLogger>>;
+	type ChannelManager = channelmanager::ChannelManager<Arc<ChainMonitor>, Arc<test_utils::TestBroadcaster>, Arc<KeysManager>, Arc<KeysManager>, Arc<KeysManager>, Arc<test_utils::TestFeeEstimator>, Arc<DefaultRouter<
+	Arc<NetworkGraph<Arc<test_utils::TestLogger>>>,
+	Arc<test_utils::TestLogger>,
+	MutexGuard<'static, TestScorer>,
```

Comment:
Hmm, I'm definitely not understanding lifetimes right here, seems like i still need a lifetime parameter despite the `for <'a> D: 'a + DerefMut<..>`

its most definitely not `'static` thats just a placeholder for now